lib/Bio/KBase/ObjectAPI/KBaseFBA/FBA.pm

```diff
@@ -3654,14 +3678,17 @@ sub parseMetaboliteInteraction {
 						}
 					}
 				}
+
 				my $thresholds = Bio::KBase::constants::auxotrophy_thresholds()->{$cpd};
-				if ($outputhash->{$cpd}->{gfrxn} >= $thresholds->[0]) {
-					$outputhash->{$cpd}->{auxotrophic} = 1;
-				}
-				my $fration = $outputhash->{$cpd}->{gfrxn}/$outputhash->{$cpd}->{totalrxn};
-				if ($fration >= $thresholds->[1]) {
-					$outputhash->{$cpd}->{auxotrophic} = 1;
-				}
+
+                if ( $thresholds && @$thresholds ) {
+
+                    my $fraction = $outputhash->{ $cpd }{ gfrxn } / $outputhash->{ $cpd }{ totalrxn };
+
+                    $outputhash->{ $cpd }{ auxotrophic } = 1
+                        if $outputhash->{ $cpd }{ gfrxn } >= $thresholds->[ 0 ]
+                        or $fraction >= $thresholds->[ 1 ];
```

Comment:
both the `if` statements in the original set `$outputhash->{ $cpd }{ auxotrophic }` to 1, so combine them into a single statement.